src/components/CopyButton.tsx

```diff
@@ -8,21 +8,37 @@ import {COLORS} from '../theme'
 export type CopyButtonProps = {
   value: string
   onCopy?: () => void
+  children?: React.ReactNode
+  style?: StyleProp<ViewStyle>
 }
 
-export const CopyButton = ({value, onCopy}: CopyButtonProps) => {
+export const CopyButton = ({value, onCopy, children, style}: CopyButtonProps) => {
   const [isCopying, copy] = useCopy()
 
   return (
-    <TouchableOpacity
-      onPress={() => {
+    <AnimatedCopyButton
+      style={style}
+      isCopying={isCopying}
+      onCopy={() => {
         copy(value)
         onCopy?.()
       }}
-      disabled={isCopying}
-      testID="copyButton"
     >
+      {children}
+    </AnimatedCopyButton>
+  )
+}
+
+const AnimatedCopyButton = ({
```

Comment:
what is this for?

Reply:
Dummy component for the copy button with animated icon when you copy

Reply:
whats a dummy component?